### DISCUSSION TITLE
Ошибки, при прослушивании подкаста

### DISCUSSION THREAD

**Discussion by Bedmit** [Chimildic/goofy#131] [Помощь с ошибкой] [answered]
Вот и дождались подкастов. Но похоже придется что то править, в связи с их появлением.
Например вот новые ошибки, при проверке условий:
`playback.context.type != 'playlist'`  -   `TypeError: Cannot read property 'type' of null`

`!playback.context.uri.includes(PLAYLIST_ID)`  -   `TypeError: Cannot read property 'uri' of null`

**Comment by Chimildic** [comment]
Можно ссылку на подкаст и полные условия _if_

**Comment by Bedmit** [comment]
Подкаст, например этот https://open.spotify.com/episode/0zSlSrh5AF2ZdqMPoWH18x?si=ND47FWUcSNu4Gnt2tQiw2Q&dl_branch=1
Такая реакция на любой подкаст. Но просто не мог пройти мимо такого названия ))

Один из скриптов - Smartify  https://github.com/Chimildic/goofy/discussions/102#discussioncomment-1345779 

Условия во втором 
```js
function LastLike() {
  const FAVORITE_IDS = ['74uSA9FulwvWqVSi7mxWyQ', '6G6y4MozfbuDLEVt2azMa6'];
  let state = Helper.readState('ActionRecomQueue');
  state.mode = state.mode || 'favorite';

  let playback = Player.getPlayback();
  if (state.mode == 'disabled'
    || playback.isEmpty()
    || playback.repeat_state == 'track'
    || playback.context.type != 'playlist'
    || (state.mode == 'favorite' && !FAVORITE_IDS.includes(Helper.parseId(playback.context.uri)))) {
    // прервать в ситуации когда нет данных, трек на паузе или повторе, играет не плейлист, не соответствие режима
    return;
  }
```

**Reply by Bedmit** [reply]
Такое впечатление, что сервис не может подкаст как-то классифицировать.

**Comment by Chimildic** [comment, marked as answer]
Нужно добавить к условию проверку наличия контекста. По-видимому его нет при включении подкаста. Потестировать не могу, интернет сбоит сегодня. 
```js
  // Первое
  if (playback.isEmpty()
    || !playback.context
    || !playback.context.uri.includes(PLAYLIST_ID)
    || playback.item.id == state.lastPlayingTrack.id
    || playback.repeat_state == 'track'
    || playback.shuffle_state) {
    return;
  } 

  // Второе
  if (state.mode == 'disabled'
    || playback.isEmpty()
    || playback.repeat_state == 'track'
    || !playback.context
    || playback.context.type != 'playlist'
    || (state.mode == 'favorite' && !FAVORITE_IDS.includes(Helper.parseId(playback.context.uri)))) {
    // прервать в ситуации когда нет данных, трек на паузе или повторе, играет не плейлист, не соответствие режима
    return;
  }
```

**Reply by Bedmit** [reply]
Фикс работает. Спасибо!
